python/dgl/heterograph.py

```diff
@@ -890,7 +890,7 @@ def remove_nodes(self, nids, ntype=None, store_ids=False):
             self._batch_num_nodes[
                 target_ntype
             ] = c_ntype_batch_num_nodes - F.astype(
-                batch_num_removed_nodes, F.int64
+                batch_num_removed_nodes, self.idtype
```

Comment:
Just wondering L884 hard-coded dtype (F.float32) is used, how do we decide when to use hard-coded, when we allow user specifying self.idtype?

Reply:
I think the underlying call `segment_reduce()` operates on float data, so it's specified as `F.float32` although `F.float64` works as well.